### PR TITLE
redirection files with conflicted document id should be built

### PR DESCRIPTION
**Pull request by OsmondJiang**
as title

fix: https://dev.azure.com/ceapex/Engineering/_workitems/edit/105471

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4878)

### CODE REVIEW

File: src/docfx/build/redirection/RedirectionMap.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static RedirectionMap Create(ErrorLog errorLog, Docset docset, Func<strin
             AddRedirections(docset.Config.Redirections, redirectDocumentId: true);
             var redirectionsByRedirectionUrl = redirections
                 .GroupBy(file => file.RedirectionUrl, PathUtility.PathComparer)
-                .ToDictionary(group => group.Key, group => group.First(), PathUtility.PathComparer);
+                .ToDictionary(group => group.Key, group => group.OrderBy(r => r).First(), PathUtility.PathComparer);
```
Comment (yufeih) [inline src/docfx/build/redirection/RedirectionMap.cs]: If we don't order here, we pick the first one that appears in redirection.json, the implication is that writers can safely add new redirection rules to the end and does not affect document_id of existing articles.

If we sort here, the above assumption is broken, new redirection rules may change document_id of an existing article. What is the expected behavior here?

File: docs/specs/config.yml [hunk 2]
```diff
@@ -493,3 +493,38 @@ outputs:
...
+    redirections:
+      a.md: /docs/path1
+      b.md: /docs/path1
+  docs/path1.md:
+outputs:
+  .errors.log: |
+    ["warning","redirection-url-conflict","The '/docs/path1' appears twice or more in the redirection mappings","docfx.yml",3,9]
+  .publish.json: |
+    {
+        "files": [
```
Comment (yufeih) [inline docs/specs/config.yml]: consider condense output expectation to only include what must be verified:

```suggestion
        "files": [
          { "url": "/a", "redirect_url": "/docs/path1" },
          { "url": "/b", "redirect_url": "/docs/path1" },
          { "url": "/docs/path1" },
```

File: src/docfx/build/redirection/RedirectionMap.cs (hunk 1 above)
Comment (OsmondJiang) [reply src/docfx/build/redirection/RedirectionMap.cs]: U r right, we should pick up the first on wrote by user

File: docs/specs/config.yml (hunk 2 above)
Comment (OsmondJiang) [reply docs/specs/config.yml]: considered, the reason why we fix it is we missed one of them in the publish items

**Comment by OsmondJiang** [comment]
@yufeih  have you updated the azure-docs-pr content which may cause this diff?

**Comment by OsmondJiang** [comment]
The diff are expected:

image

**Pull request merged**